rust/src/example_iteration.rs

```diff
@@ -117,16 +123,21 @@ struct ShardProgress {
 
 /// Return a vector of bytes with the file content.
 fn get_file_bytes(shard_info: &ShardInfo) -> Vec<u8> {
+    let mut file_bytes = Vec::new();
     match shard_info.compression_type {
-        CompressionType::Uncompressed => std::fs::read(&shard_info.file_path).unwrap(),
+        CompressionType::Uncompressed => return std::fs::read(&shard_info.file_path).unwrap(),
         CompressionType::LZ4 => {
-            let mut file_bytes = Vec::new();
             lz4_flex::frame::FrameDecoder::new(std::fs::File::open(&shard_info.file_path).unwrap())
                 .read_to_end(&mut file_bytes)
                 .unwrap();
-            file_bytes
         }
-    }
+        CompressionType::Gzip | CompressionType::Zlib => {
+            flate2::read::GzDecoder::new(std::fs::File::open(&shard_info.file_path).unwrap())
+                .read_to_end(&mut file_bytes)
+                .unwrap();
```

Comment:
nit (depending on how many compression algorithms you'll have): you can use a helper function for branches that produce a `Read` object.

```rust
fn read_to_end(mut reader: impl std::io::Read) -> Result<Vec<u8>> {
    let mut result = Vec::new();
    reader.read_to_end(&mut result)?;
    Ok(result)
}
```

```suggestion
            read_to_end(flate2::read::GzDecoder::new(std::fs::File::open(&shard_info.file_path).unwrap()))
```